runelite-client/src/main/java/net/runelite/client/plugins/poh/PohPlugin.java

```diff
@@ -145,6 +197,67 @@ public void onGameStateChanged(GameStateChanged event)
 		if (event.getGameState() == GameState.LOADING)
 		{
 			pohObjects.clear();
+			incenseBurners.clear();
 		}
 	}
-}
+
+	@Subscribe
+	public void onAnimationChanged(AnimationChanged event)
+	{
+		final Actor actor = event.getActor();
+		final String actorName = actor.getName();
+
+		if (!(actor instanceof Player) || actor.getAnimation() != AnimationID.INCENSE_BURNER)
+		{
+			return;
+		}
+
+		final int x = actor.getLocalLocation().getX();
+		final int y = actor.getLocalLocation().getY();
+
+		// Find burner closest to player
+		incenseBurners.keySet().stream().min((a, b) -> ComparisonChain.start()
+			.compare(Math.abs(a.getLocalLocation().getY() - y), Math.abs(b.getLocalLocation().getY() - y))
+			.compare(Math.abs(a.getLocalLocation().getX() - x), Math.abs(b.getLocalLocation().getX() - x))
+			.result())
+			.ifPresent(tile ->
+			{
+				final IncenseBurner incenseBurner = incenseBurners.get(tile);
+
+				if (actor == client.getLocalPlayer())
+				{
+					int level = client.getRealSkillLevel(net.runelite.api.Skill.FIREMAKING);
+					updateBurner(incenseBurner, level);
+				}
+				else if (actorName != null)
+				{
+					lookupPlayer(actorName, incenseBurner);
+				}
+			});
+
+	}
+
+	private void lookupPlayer(String playerName, IncenseBurner incenseBurner)
+	{
+		executor.execute(() ->
+		{
+			try
+			{
+				final HiscoreResult playerStats = hiscoreManager.lookup(playerName, HiscoreEndpoint.NORMAL);
+				final Skill fm = playerStats.getFiremaking();
```

Comment:
`https://github.com/runelite/runelite/commit/1f8ce36688061becc5c38c724724fd9ac88fd255` logs errors from executor jobs now I think so you will probably want to error check all of this